app/controllers/admin/api/services/proxy/policies_controller.rb

```diff
@@ -56,6 +56,11 @@ def authorize_rolling_update
 
   def proxy_params
     proxy_params = params.require(:proxy).dup
+    policies_config_params = proxy_params[:policies_config]
+
+    if policies_config_params && policies_config_params.is_a?(ActionController::Parameters) && policies_config_params.to_unsafe_h.is_a?(Hash)
```

Comment:
This makes it too much responsibility

You can create a wrapper to transform the `policies_config_params` to an object you know the behaviour ;)

Reply:
Some extra info from Hery:

> this check should not be in the controller action itself
> you can create an object that makes those check for you and spit out the desired format (Array or Hash)
> And then you do not even need a is_a? I think :slightly_smiling_face: